### PR TITLE
Save all screenshots and page sources + show their names in log message

### PR DESCRIPTION
**Pull request by kusnir**
See https://github.com/Codeception/Codeception/pull/3490

### CODE REVIEW

File: src/Codeception/Module/WebDriver.php [hunk 1]
```diff
@@ -374,9 +374,18 @@ public function _failed(TestInterface $test, $fail)
...
-        $this->_saveScreenshot($outputDir . mb_strcut($filename, 0, 245, 'utf-8') . '.fail.png');
-        $this->_savePageSource($outputDir . mb_strcut($filename, 0, 244, 'utf-8') . '.fail.html');
-        $this->debug("Screenshot and page source were saved into '$outputDir' dir");
+        $filenamePng = mb_strcut($filename, 0, 245, 'utf-8') . '.fail.png';
+        $filenameHtml = mb_strcut($filename, 0, 244, 'utf-8') . '.fail.html';
+        for ($i = 1; $i < 1000; $i++) {
+            if ((! file_exists($outputDir . $filenamePng)) && (! file_exists($outputDir . $filenameHtml))) {
+                break;
+            }
+            $filenamePng = mb_strcut($filename . '.' . $i, 0, 245, 'utf-8') . '.fail.png';
```
Comment (Naktibalda) [inline src/Codeception/Module/WebDriver.php]: Append $i after strcut or it will be lost if $filename is too long.

Comment (Naktibalda) [inline src/Codeception/Module/WebDriver.php]:
```php
$filenamePng = mb_strcut($filename, 0, 244 - strlen($i), 'utf-8') . ".$i.fail.png";
```

File: src/Codeception/Module/WebDriver.php [hunk 2]
```diff
@@ -374,9 +374,18 @@ public function _failed(TestInterface $test, $fail)
         $this->debugWebDriverLogs();
         $filename = preg_replace('~\W~', '.', Descriptor::getTestSignature($test));
         $outputDir = codecept_output_dir();
-        $this->_saveScreenshot($outputDir . mb_strcut($filename, 0, 245, 'utf-8') . '.fail.png');
-        $this->_savePageSource($outputDir . mb_strcut($filename, 0, 244, 'utf-8') . '.fail.html');
-        $this->debug("Screenshot and page source were saved into '$outputDir' dir");
+        $filenamePng = mb_strcut($filename, 0, 245, 'utf-8') . '.fail.png';
+        $filenameHtml = mb_strcut($filename, 0, 244, 'utf-8') . '.fail.html';
+        for ($i = 1; $i < 1000; $i++) {
+            if ((! file_exists($outputDir . $filenamePng)) && (! file_exists($outputDir . $filenameHtml))) {
```
Comment (DavertMik) [inline src/Codeception/Module/WebDriver.php]: actually this is a mess. No one will ever be able to guess which files belong to which build unless looking to their date. Current approach is probably not the best as well but at least there is only 1 file and not 1000 of them.

**Comment by DavertMik** [comment]
This can't be accepted unless we change policy for naming reports.

Let's say we have 3 failed files from 1st run, 5 failed from 2nd and 1 from 3rd. Then by looking into them we will have no ideas what and when actually went wrong. 

If we add timestamp to filenames we will have dozens of screenshots created in a few runs. 

Current behavior is stupid, dead simple, and effective. I can't think of something better for now :( 
This improvement can be moved to extension (the similar stuff is done via RecorderExtension). 

Can you take the https://github.com/Codeception/Codeception/blob/2.2/ext/Recorder.php and make it worked for failing steps only and not for all steps. Then you can print to console failed step names + corresponding file names.

**Comment by DavertMik** [comment]
Sorry, but I need to close this PR.
@kusnir I recommend you to create an extension, publish it on packagist and add it as [Addon](http://codeception.com/addons) on our site. This will make this functionality work. 
Thank you for submitting PR. Sorry we can't accept this one.

**Pull request closed without merging**